model_compression_toolkit/target_platform_capabilities/schema/v1.py

```diff
@@ -0,0 +1,727 @@
+            return False
+
+        # Check for containment by comparing operator groups
+        for i in range(len(self.operator_groups_list) - len(other.operator_groups_list) + 1):
+            for j in range(len(other.operator_groups_list)):
+                if self.operator_groups_list[i + j] != other.operator_groups_list[j] and not (
+                        isinstance(self.operator_groups_list[i + j], OperatorSetConcat) and (
+                        other.operator_groups_list[j] in self.operator_groups_list[i + j].op_set_list)):
+                    break
+            else:
+                # If all checks pass, the other Fusing instance is contained
+                return True
+        # Other Fusing instance is not contained
+        return False
+
+    def get_info(self):
+        """
+        Retrieves information about the Fusing instance, including its name and the sequence of operator groups.
+
+        Returns:
+            A dictionary with the Fusing instance's name as the key and the sequence of operator groups as the value,
+            or just the sequence of operator groups if no name is set.
+        """
+        if self.name is not None:
+            return {self.name: ' -> '.join([x.name for x in self.operator_groups_list])}
+        return ' -> '.join([x.name for x in self.operator_groups_list])
+
+
+class TargetPlatformModel(ImmutableClass):
+    """
+    Represents the hardware configuration used for quantized model inference.
+
+    This model defines:
+    - The operators and their associated quantization configurations.
+    - Fusing patterns, enabling multiple operators to be combined into a single operator
+      for optimization during inference.
+    - Versioning support through minor and patch versions for backward compatibility.
+
+    Attributes:
+        SCHEMA_VERSION (int): The schema version of the target platform model.
+    """
+    SCHEMA_VERSION = 1
+    def __init__(self,
+                 default_qco: QuantizationConfigOptions,
+                 tpc_minor_version: Optional[int],
+                 tpc_patch_version: Optional[int],
+                 add_metadata: bool = True,
+                 name="default_tp_model"):
+        """
+
+        Args:
+            default_qco (QuantizationConfigOptions): Default QuantizationConfigOptions to use for operators that their QuantizationConfigOptions are not defined in the model.
+            tpc_minor_version (Optional[int]): The minor version of the target platform capabilities.
+            tpc_patch_version (Optional[int]): The patch version of the target platform capabilities.
+            add_metadata (bool): Whether to add metadata to the model or not.
+            name (str): Name of the model.
+
+         Raises:
+            AssertionError: If the provided `default_qco` does not contain exactly one quantization configuration.
+        """
+
+        super().__init__()
+        self.tpc_minor_version = tpc_minor_version
```

Comment:
need to add another foeld to the versioning scheme - "platform_type" (e.g., imx500, aisoc...).
make it a string and add it to the metadata as well